cpp/cmake/3rdparty.cmake

```diff
@@ -0,0 +1,94 @@
+find_package(Git QUIET)
+
+if(NOT ${CODING_CONV_PREFIX}_3RDPARTY_DIR)
+  set(${CODING_CONV_PREFIX}_3RDPARTY_DIR 3rdparty)
+  set(${CODING_CONV_PREFIX}_3RDPARTY_DIR
+      3rdparty
+      PARENT_SCOPE)
+endif()
+
+# initialize submodule with given path
+function(bbp_init_git_submodule path)
+  if(NOT ${GIT_FOUND})
+    message(
+      FATAL_ERROR "git not found and ${path} submodule not cloned (use git clone --recursive)")
+  endif()
+  message(
+    STATUS "Fetching git submodule ${path}: running git submodule update --init --recursive ${path}"
+  )
+  execute_process(
+    COMMAND
+      ${GIT_EXECUTABLE} submodule update --init --recursive -- ${path}
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    RESULT_VARIABLE git_submodule_status)
+  if(NOT git_submodule_status EQUAL 0)
+    message(FATAL_ERROR "Could not clone git submodule ${path}")
+  endif()
+endfunction()
+
+# use a git submodule
+#
+# bbp_git_submodule(source_dir
+#                   [DISABLED]
+#                   SUBDIR path
+#                   [BUILD] [<arguments>]
+#                   [PACKAGE] [<arguments>]
+#
+# Add a CMake option in the cache to control whether the
+# submodule is used or not (default ON). The option is named after the source
+# directory passed in first argument, for instance:
+#   bbp_git_submodule(src/eigen)
+# adds the following CMake cached option:
+#  ${PROJECT_NAME}_3RDPARTY_USE_SRC_EIGEN:BOOL=ON
+#
+# If enabled, then the submodule is fetched if missing in the working copy.
+#
+# If the DISABLED argument is provided, then the default value for the CMake
+# option is OFF.
+#
+# if the BUILD argument is provided then the directory is added to the build
+# through the add_subdirectory CMake function. Arguments following the BUILD
+# arguments are passed to the add_subdirectory function call.
+#
+# The optional SUBDIR argument is used by the BUILD argument to determine
+# the path to the directory added to the build. The path specified is relative
+# to the path to the git submodule.
+#
+# If the PACKAGE argument is provided and the CMake option to determine whether
+# the git submodule should be used or not is FALSE, then a call to the find_package
+# function is made with the arguments specified to the PACKAGE option.
+#
+function(bbp_git_submodule name)
```

Comment:
(very) minor : I am thinking about the `bbp` part in the function name: we use this in non-bbp organisation repos like neuronsimulator/nrn. I am wondering if we can have something else in the prefix. (But it's perfectly fine if we keep it like this!)

Reply:
This project implements the C++ Coding Convention used by the Blue Brain Project. I still would like to prefix the symbols exposed. I have no problem changing the prefix. What about cppcc (C++ Coding Conventions)?
Any other idea?

Reply:
I don't have any better suggestion, either option is fine (`cpp_cc` might be better?).

Reply:
I updated the prefix of all CMake functions from `bbp_` to `cpp_cc` in 33870967